invisible_cities/cities/beersheba_test.py

```diff
@@ -103,3 +107,23 @@ def test_beersheba_param_dim(ICDATADIR, deconvolution_config, ndim):
 
     with raises(ValueError):
         beersheba(**conf)
+
+
+@mark.parametrize("param_name", ('cut_type', 'deconv_mode', 'energy_type', 'inter_method'))
+def test_deconvolve_signal_enums(ICDATADIR, deconvolution_config, param_name):
+    true_out         = os.path.join(ICDATADIR, "test_Xe2nu_NEW_exact_deconvolution_joint.h5")
+    conf, PATH_OUT   = deconvolution_config
+    conf_dict        = conf['deconv_params']
+
+    conf_dict.pop("q_cut")
+    conf_dict.pop("drop_dist")
+
+    conf_dict['cut_type'    ] = CutType            (conf_dict['cut_type'    ])
+    conf_dict['deconv_mode' ] = DeconvolutionMode  (conf_dict['deconv_mode' ])
+    conf_dict['energy_type' ] = HitEnergy          (conf_dict['energy_type' ])
+    conf_dict['inter_method'] = InterpolationMethod(conf_dict['inter_method'])
+
+    conf_dict[param_name]     = 'check'
```

Comment:
```suggestion
    conf_dict[param_name]     = param_name
```
It is particularly interesting to check that it doesn't work even if you try to use the name of the parameter, as it is the most-likely error.